src/riak_kv_app.erl

```diff
@@ -49,7 +49,21 @@
 start(_Type, _StartArgs) ->
     riak_core_util:start_app_deps(riak_kv),
 
-    FSM_Limit = app_helper:get_env(riak_kv, fsm_limit, ?DEFAULT_FSM_LIMIT),
+    FSM_Limit = case app_helper:get_env(riak_kv, fsm_limit, ?DEFAULT_FSM_LIMIT) of
+                    undefined ->
+                        %% If it's explicitly set to 'undefined', that means "no limit".
+                        %% sidejob doesn't have an option to specify "infinity" but if
+                        %% we make the limit equal to the node's global process limit,
+                        %% then we're pretty much guaranteed to never exceed that:
+                        erlang:system_info(process_limit);
```

Comment:
To be honest, it's no more _dangerous_ then infinite has always been.  The user would be specifically turning off the governor.  I'm not sure if there is a good way to know or even estimate the number of processes necessary to allow the node not to get sad when you are running out this far.   I'm for safer where we can get it, but this option has always been a bit "Hey Y'all, Watch this..."